cluster.go

```diff
@@ -1135,8 +1127,9 @@ func (c *Cluster) allocate(hash *cid.Cid, repl int) ([]peer.ID, error) {
 
 		// we don't have enough peers to pin
 		if len(candidateAllocs) < needed {
-			err = logError("cannot find enough allocations for this CID: needed: %d. Got: %s",
-				needed, candidateAllocs)
+			err = logError(
+				"cannot find enough allocations for %s. Needed: %d. Got: %d",
+				hash, needed, candidateAllocs)
```

Comment:
Might be nice to print out the number as well (e.g. `Got <len(candidateAllocs>: <candidateAllocs>`). But otherwise, this all lgtm :+1: